mirrord/operator/src/crd.rs

```diff
@@ -199,3 +341,190 @@ pub struct MirrordPolicySpec {
     /// List of features and operations blocked by this policy.
     pub block: Vec<BlockedFeature>,
 }
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize, JsonSchema)]
+#[serde(rename_all = "camelCase")] // queue_name_key -> queueNameKey in yaml.
+pub struct ConfigMapQueueNameSource {
+    /// The name of the config map that holds the name of the queue we want to split.
+    pub name: String,
+
+    /// The name of the key in the config map that holds the name of the queue we want to
+    /// split.
+    pub queue_name_key: String,
+}
+
+/// Set where the application reads the name of the queue from, so that mirrord can find that queue,
+/// split it, and temporarily change the name there to the name of the branch queue when splitting.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize, JsonSchema)]
+#[serde(rename_all = "camelCase")] // ConfigMap -> configMap in yaml.
+pub enum QueueNameSource {
+    ConfigMap(ConfigMapQueueNameSource),
+    EnvVar(String),
+}
+
+pub type OutputQueueName = String;
+
+/// The details of a queue that should be split.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize, JsonSchema)]
+#[serde(tag = "queueType")]
+// So that controllers that only handle 1 type of queue don't have to adapt when we add more queue
+// types.
+#[non_exhaustive]
```

Comment:
I'm not a big fan of this, but I won't complain :v

Reply:
What's the problem with that?